cli-toolchain/graphql.md

```diff
@@ -912,13 +914,123 @@ mutation CreateDraft {
 }
 ```
 
+#### `public` Authorization
+
+```
+# The simplest case
+type Post @model @auth(rules: [{allow: public}]) {
+  id: ID!
+  title: String!
+}
+```
+
+The `public` authorization specifies that everyone will be allowed to access the API, behind the scenes the API will be protected with an API Key. To be able to use `public` the API must have API Key configured.
+
+```
+# public authorization with provider override
+type Post @model @auth(rules: [{allow: public, provider: iam}]) {
+  id: ID!
+  title: String!
+}
+```
+
+The `@auth` directive allows the override of the default provider for a given authorization mode. In the sample above `iam` is specified as the provider. The CLI generates the necessary IAM policies for the `unauthenticated` role automatically.
+
+#### `private` Authorization
+
+```
+# The simplest case
+type Post @model @auth(rules: [{allow: private}]) {
+  id: ID!
+  title: String!
+}
+```
+
+The `private` authorization specifies that everyone will be allowed to access the API with a valid JWT token from the configured Cognito User Pool. To be able to use `private` the API must have Cognito User Pool configured.
+
+```
+# private authorization with provider override
+type Post @model @auth(rules: [{allow: private, provider: iam}]) {
+  id: ID!
+  title: String!
+}
+```
+
+The `@auth` directive allows the override of the default provider for a given authorization mode. In the sample above `iam` is specified as the provider. The CLI generates the necessary IAM policies for the `authenticated` role automatically.
```

Comment:
Similar to feedback from above can you reword this for clarification:

The `@auth` directive allows the override of the default provider for a given authorization mode. In the sample above iam is specified as the provider which allows you to use an "Authenticated Role" from Cognito Identity Pools for private access. When used in conjunction with `amplify add auth` the CLI generates scoped down IAM policies for the "Authenticated" role automatically.

Reply:
Changed.